Cython/Compiler/ModuleNode.py

```diff
@@ -3255,8 +3255,10 @@ def generate_module_import_setup(self, env, code):
             # out the module path at runtime by rerunning the import lookup
             code.putln("if (!CYTHON_PEP489_MULTI_PHASE_INIT) {")
             package_name, _ = self.full_module_name.rsplit('.', 1)
+            module_name = env.module_name
             if '.' in package_name:
                 parent_name = '"%s"' % (package_name.rsplit('.', 1)[0],)
+                module_name = '.' + module_name
```

Comment:
This seems worth explaining in a code comment. It's not obvious at all why a module name would need prefixing with a dot. Is this triggering a relative import?

Reply:
The C code calls the equivalent of `importlib.util.find_spec(module_name, parent_name)`. The documentation [for `find_spec`](https://docs.python.org/3.8/library/importlib.html#importlib.util.find_spec) mentions that the semantics of its arguments are the same as [for `imprt_module`](https://docs.python.org/3.8/library/importlib.html#importlib.import_module), which says 
> If the name is specified in relative terms, then the package argument must be set to the name of the package which is to act as the anchor for resolving the package name (e.g. `import_module('..mod', 'pkg.subpkg')` will import `pkg.mod`). 

This all can be simplified now that it uses `find_spec` to use the `full_module_name`, without a package at all.

Fixing.